app/validators/uri_validator.rb

```diff
@@ -0,0 +1,32 @@
+# frozen_string_literal: true
+
+class UriValidator < ActiveModel::EachValidator
+  DEFAULT_ACCEPTED_SCHEMES = /^https?$/
+  DEFAULT_OPTIONAL_PARTS = %i[port].freeze
+  DEFAULT_FORBIDDEN_PARTS = %i[userinfo registry path opaque query fragment].freeze
+
+  def validate_each(record, attribute, value)
+    uri = URI.parse(value)
+    raise URI::InvalidURIError if !valid_scheme?(uri.scheme) || uri.host.blank? || forbidden_part?(uri)
+    true
+  rescue URI::InvalidURIError
+    record.errors.add(attribute, options[:message] || :invalid)
+    false
+  end
+
+  def valid_scheme?(scheme)
+    accepted_scheme = options.fetch(:scheme, DEFAULT_ACCEPTED_SCHEMES)
+    case accepted_scheme
+    when Regexp
+      scheme =~ accepted_scheme
+    else
+      [*accepted_scheme].include? scheme
```

Comment:
So it can be an Array or a Regexp?
Why not using only an Array?

Reply:
I think it's a matter of taste more than anything. I was thinking on the usage:

```ruby
validates :field, uri: { scheme: /^[st]?ftp$/ }
```

instead of 

```ruby
validates :field, uri: { scheme: %w[ftp sftp tftp] }
```

But I can change that if you think it's worth. No problem.

Reply:
This list is not changing so much, you just add complexity here I think